.github/workflows/macos_job.yml

```diff
@@ -0,0 +1,119 @@
+name: Run a macOS job
+
+on:
+  workflow_call:
+    inputs:
+      script:
+        description: 'Script to utilize'
+        default: "python setup.py bdist_wheel"
+        type: string
+      timeout:
+        description: 'Timeout for the job (in minutes)'
+        default: 30
+        type: number
+      runner:
+        description: 'Runner type to utilize'
+        default: "macos-12"
+        type: string
+      upload-artifact:
+        description: 'Name to give artifacts uploaded from ${RUNNER_ARTIFACT_DIR}'
+        default: ""
+        type: string
+      download-artifact:
+        description: 'Name to download artifacts to ${RUNNER_ARTIFACT_DIR}'
+        default: ""
+        type: string
+      repository:
+        description: 'Repository to checkout, defaults to ""'
```

Comment:
nit: not immediately clear what the default value of '' represents: whether is it caller's repo or "no checkout".

Reply:
Looks like it's caller's repo.